standardtracer/spanpropagator.go

```diff
@@ -0,0 +1,194 @@
+package standardtracer
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/opentracing/opentracing-go"
+)
+
+const (
+	fieldNameTraceID = "traceid"
+	fieldNameSpanID  = "spanid"
+	fieldNameSampled = "sampled"
+)
+
+func (s *tracerImpl) PropagateSpanAsText(
+	sp opentracing.Span,
+) (
+	contextIDMap map[string]string,
+	attrsMap map[string]string,
+) {
+	sc := sp.(*spanImpl).raw.StandardContext
+	contextIDMap = map[string]string{
+		fieldNameTraceID: strconv.FormatInt(sc.TraceID, 10),
+		fieldNameSpanID:  strconv.FormatInt(sc.SpanID, 10),
+		fieldNameSampled: strconv.FormatBool(sc.Sampled),
+	}
+	sc.tagLock.RLock()
+	attrsMap = make(map[string]string, len(sc.traceAttrs))
+	for k, v := range sc.traceAttrs {
+		attrsMap[k] = v
+	}
+	sc.tagLock.RUnlock()
+	return contextIDMap, attrsMap
+}
+
+func (s *tracerImpl) PropagateSpanAsBinary(
+	sp opentracing.Span,
+) (
+	traceContextID []byte,
+	traceAttrs []byte,
+) {
+	sc := sp.(*spanImpl).raw.StandardContext
+	var err error
+
+	// Handle the trace and span ids.
+	contextBuf := new(bytes.Buffer)
+	err = binary.Write(contextBuf, binary.BigEndian, sc.TraceID)
+	if err != nil {
+		panic(err)
+	}
+	err = binary.Write(contextBuf, binary.BigEndian, sc.SpanID)
+	if err != nil {
+		panic(err)
+	}
+
+	// Handle the attributes.
```

Comment:
FYI, I haven't tested this yet and so it probably has some subtle bug. I'll write a unittest once we agree that trace attrs are here to stay. :)